### PR TITLE
Completed Challenge 122

### PR DESCRIPTION
**Pull request by gaebar**
I have created a new branch for challenge 122, as I have based the previous branch 122 upon the wrong base. See the full history here: https://github.com/gaebar/python-by-example-exercises/commits/challenge-122

### CODE REVIEW

File: exercises/subprograms/challenge_122.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    salary = get_user_input_float("Enter salary: £")
+
+    locale.setlocale(locale.LC_ALL, "en_GB")
+
+    # Using the locale built in library and CSV docwriter library.
+    # Format the salary number with the currency symbol and grouping the
+    # thousands.
+    with open(CSV_FILE_NAME, "a", newline="") as csvfile:
+        writer = csv.DictWriter(csvfile, fieldnames=FIELD_NAMES)
+        writer.writerow({FIELD_NAMES[0]: name, FIELD_NAMES[1]: locale.currency(salary, True, True)})
```
Comment (jneves) [inline exercises/subprograms/challenge_122.py]: As a general rule, always save a number as a number when you are storing it. Leave the locale formatting for showing it to the user. This allows you to have the numbers in a consistent format and present the same numbers to users that choose to see different locale/currencies.

**Review by jneves — CHANGES_REQUESTED** [review]
Why are the tests missing? Forgotten file?

File: exercises/subprograms/challenge_122.py (hunk 1 above)
Comment (gaebar) [reply exercises/subprograms/challenge_122.py]: This was a test for me to try the locale format library. I have updated now.

**Comment by gaebar** [comment]
I've added a test for this challenge. Let me know if it looks good

**Review by jneves — APPROVED** [review]
Looks good, the test could be better by using mocks to catch the writes though. Something to add to your list of things to learn.

**Pull request merged**